Fill.py

```diff
@@ -136,36 +136,34 @@ def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations:
     itempool.extend(unplaced_items)
 
 
-def fast_fill(world: MultiWorld, locations: typing.List[Location],
-                     itempool: typing.List[Item], single_player_placement: bool = False) -> None:
+def remaining_fill(world: MultiWorld,
+                   locations: typing.List[Location],
+                   itempool: typing.List[Item]) -> None:
     unplaced_items: typing.List[Item] = []
     placements: typing.List[Location] = []
-    swap = 0
     swapped_items: typing.Counter[typing.Tuple[int, str]] = Counter()
     while locations and itempool:
         item_to_place = itempool.pop()
         spot_to_fill: typing.Optional[Location] = None
 
         for i, location in enumerate(locations):
-            if location.item_rule(item_to_place) and (not single_player_placement or
-                                                      location.player == item_to_place.player):
+            if location.item_rule(item_to_place):
                 # popping by index is faster than removing by content,
                 spot_to_fill = locations.pop(i)
                 # skipping a scan for the element
                 break
 
         else:
-            swap += 1
             # we filled all reachable spots.
             # try swapping this item with previously placed items
 
             for (i, location) in enumerate(placements):
                 placed_item = location.item
                 # Unplaceable items can sometimes be swapped infinitely. Limit the
                 # number of times we will swap an individual item to prevent this
-                swap_count = swapped_items[placed_item.player,
-                                           placed_item.name]
-                if swap_count > 1 or placed_item.advancement:
```

Comment:
We need to ensure it does not swap out advancement items so they do not get placed in this routine which disregards access rules

Reply:
But location.item comes from placements and placements is filled within this function from locations only, which are all empty at the beginning of this call, so all contents sshould never be advancement

Reply:
Oh right. I knew that. I was just seeing if you knew